casper/src/slowcooker/scala/coop.rchain.casper/HashSetCasperSpecification.scala

```diff
@@ -1,247 +1,229 @@
-//
-//    override def run(sut: Sut): Result = {
-//      val validator = sut.nodes(node.idx).node
-//      val result = (deploy(validator, deployment(0)) >> create(validator) >>= (
-//          b => add(validator, b)
-//      )).result.right.get
-//      (result, validator.logEff.errors ++ validator.logEff.warns)
-//    }
-//
-//    override def nextState(state: State): State = state
-//
-//    override def preCondition(state: State): Boolean = !state(node.idx).deployed
-//
-//    override def postCondition(state: State, result: Try[Result]): Prop = isValidAndNoErrors(result)
-//  }
-//
-//  case class Receive(node: RNode) extends Command {
-//    override type Result = List[String]
-//
-//    override def run(sut: Sut): List[String] = {
-//      val validator = sut.nodes(node.idx).node
-//      validator.receive().result
-//      validator.logEff.errors ++ validator.logEff.warns
-//    }
-//
-//    override def nextState(state: State): State = state
-//
-//    override def preCondition(state: State): Boolean = true
-//
-//    override def postCondition(state: State, result: Try[Result]): Prop =
-//      result.isSuccess && result.get.isEmpty
-//  }
-//}
+package coop.rchain.casper
+
+import scala.util.{Random, Try}
+import cats.effect.Sync
+import cats.implicits._
+import com.google.protobuf.ByteString
+import coop.rchain.blockstorage.dag.BlockDagStorage.DeployId
+import coop.rchain.blockstorage.syntax._
+import coop.rchain.casper.MultiParentCasper.ignoreDoppelgangerCheck
+import coop.rchain.casper.blocks.proposer.Created
+import coop.rchain.casper.genesis.Genesis
+import coop.rchain.casper.genesis.contracts._
+import coop.rchain.casper.helper.TestNode
+import coop.rchain.casper.helper.TestNode._
+import coop.rchain.casper.protocol.{BlockMessage, DeployData}
+import coop.rchain.casper.util.{ConstructDeploy, GenesisBuilder}
+import coop.rchain.catscontrib.TaskContrib._
+import coop.rchain.crypto.signatures.{Secp256k1, Signed}
+import coop.rchain.crypto.PublicKey
+import coop.rchain.models.BlockHash
+import coop.rchain.rholang.interpreter.util.RevAddress
+import monix.eval.Task
+import monix.execution.Scheduler.Implicits.global
+import org.scalacheck._
+import org.scalacheck.commands.Commands
+
+case class RNode(idx: Int, name: String, deployed: Boolean)
+
+object HashSetCasperProperties extends Properties("HashSetCasper") {
+  implicit def noShrink[T]: Shrink[T] = Shrink.shrinkAny
+  property("network") = HashSetCasperSpecification.property()
+}
+
+object HashSetCasperPropertiesApp {
+  def main(args: Array[String]): Unit =
+    HashSetCasperSpecification.property().check(_.withMinSuccessfulTests(1))
+}
+
+class NodeBox(val node: TestNode[Effect], var lastBlock: Option[BlockMessage]) {
+  def update(bm: Option[BlockMessage]): Unit = this.lastBlock = bm
+}
+
+object HashSetCasperActions {
+  import GenesisBuilder._
+
+  def context(
+      amount: Int,
+      bondsGen: Seq[PublicKey] => Map[PublicKey, Long]
```

Comment:
`bondsGen` is not used.

Reply:
It's old code and not used, do you want me to fix it or leave it as it is?

Reply:
https://github.com/rchain/rchain/pull/3355/commits/fe99286a4e6aade733a95e754960f688475dc21e